src/ui/components/LedgerAdmin.js

```diff
@@ -74,38 +87,54 @@ export const LedgerAdmin = (): ReactNode => {
     let nextLedger;
     if (ledger.account(id).active) {
       nextLedger = ledger.deactivate(id);
-      setCheckBoxSelected(false);
     } else {
       nextLedger = ledger.activate(id);
-      setCheckBoxSelected(true);
     }
     updateLedger(nextLedger);
   };
 
   const resetIdentity = () => {
     setIdentityName("");
     setSelectedId(null);
-    setCheckBoxSelected(false);
     setPromptString("Add Identity: ");
   };
 
   const setActiveIdentity = (identity: Identity) => {
     setIdentityName(identity.name);
     setSelectedId(identity.id);
-    setCheckBoxSelected(ledger.account(identity.id).active);
     setPromptString("Update Identity: ");
   };
 
+  const loadEntitiesCheckList = () => {
+    const identities = {};
+
+    ledger.accounts().map(({identity, active}) => {
+      identities[`${identity.id}`] = active;
+    });
+
+    setIsChecked(identities);
+  };
+
+  useEffect(() => loadEntitiesCheckList(), [ledger]);
```

Comment:
No need to define the function outside of useEffect

```suggestion
  useEffect(() => {
    const identities = {};

    ledger.accounts().map(({identity, active}) => {
      identities[`${identity.id}`] = active;
    });

    setIsChecked(identities);
  }, [ledger]);
```